src/studio/src/designer/backend/Services/Implementation/SchemaModelService.cs

```diff
@@ -56,23 +69,151 @@ public async Task UpdateSchema(string org, string repository, string developer,
                 var schemaName = GetSchemaName(relativeFilePath);
-                var jsonSchema = await DeserializeJson(jsonContent);
-                var rootName = GetRootName(jsonSchema);
-
-                await UpdateJsonSchema(altinnAppGitRepository, relativeFilePath, jsonContent);
-                await UpdateXsd(altinnAppGitRepository, jsonSchema, schemaName);
-                var modelMetadata = await UpdateModelMetadata(altinnAppGitRepository, jsonSchema, schemaName);
-                await UpdateApplicationMetadata(altinnAppGitRepository, schemaName, rootName);
-                await UpdateCSharpClasses(altinnAppGitRepository, modelMetadata, schemaName);
+
+                await DeleteDatatypeFromApplicationMetadata(altinnAppGitRepository, schemaName);
+                DeleteRelatedSchemaFiles(altinnAppGitRepository, schemaName, altinnCoreFile.Directory);
+            }
+            else
+            {
+                altinnGitRepository.DeleteFileByRelativePath(relativeFilePath);
+            }
+        }
+
+        private async Task UpdateAllAppModelFiles(string org, string repository, string developer, string relativeFilePath, string jsonContent)
+        {
+            var altinnAppGitRepository = _altinnGitRepositoryFactory.GetAltinnAppGitRepository(org, repository, developer);
+
+            var schemaName = GetSchemaName(relativeFilePath);
+            var jsonSchema = await DeserializeJson(jsonContent);
+            var rootName = GetRootName(jsonSchema);
+
+            await UpdateJsonSchema(altinnAppGitRepository, relativeFilePath, jsonContent);
+            await UpdateXsd(altinnAppGitRepository, jsonSchema, schemaName);
+            var modelMetadata = await UpdateModelMetadata(altinnAppGitRepository, jsonSchema, schemaName);
+            await UpdateApplicationMetadata(altinnAppGitRepository, schemaName, rootName);
+            await UpdateCSharpClasses(altinnAppGitRepository, modelMetadata, schemaName);
+        }
+
+        private async Task UpdateAppTexts(string org, string repository, string developer, JsonSchema jsonSchema)
+        {
+            JsonSchemaToInstanceModelGenerator converter = new JsonSchemaToInstanceModelGenerator(org, repository, jsonSchema);
+            var newTexts = converter.GetTexts();
+
+            var altinnAppGitRepository = _altinnGitRepositoryFactory.GetAltinnAppGitRepository(org, repository, developer);
+            var existingTexts = await altinnAppGitRepository.GetTextResourcesForAllLanguages();
+
+            MergeTexts(newTexts, existingTexts);
+
+            await altinnAppGitRepository.SaveServiceTexts(existingTexts);
+        }
+
+        private JsonSchema GenerateJsonSchema(Stream xsdStream)
+        {
+            var xmlReader = XmlReader.Create(xsdStream, new XmlReaderSettings { IgnoreWhitespace = true });
+            var xsdToJsonSchemaConverter = new XsdToJsonSchema(xmlReader, _loggerFactory.CreateLogger<XsdToJsonSchema>());
```

Comment:
are we risking a null reference here if the alternative contructor is utilized?

Reply:
Yes, we are. It was a lazy man's doing when I wrote some tests (it's only used by tests as the logger is injected by the container). But I'll fix it and remove that constructor.